lips/lip-20.md

```diff
@@ -0,0 +1,425 @@
+
+## <a name="currency_ops">Currency Operations</a>
+
+There are a number of different operations that may be performed to view and update the
+metadata for a given currency (viz. [Required information for registration of a currency
+and metadata](#metadata_spec)). They can be divided in to "getters" (non-mutative),
+"setters" (mutative), and predicate operations, and are as follows:
+
+### Non-Mutative ("getters")
+* [`Libra::preburn_value<CoinType>(): u64`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-preburn_value)
+* [`Libra::market_cap<CoinType>(): u128`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-market_cap)
+* [`Libra::approx_lbr_for_value<FromCoinType>(from_value: u64): u64`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-approx_lbr_for_value)
+* [`Libra::approx_lbr_for_coin<FromCoinType>(coin: &Libra<FromCoinType>): u64`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-approx_lbr_for_coin)
+* [`Libra::scaling_factor<CoinType>(): u64`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-scaling_factor)
+* [`Libra::fractional_part<CoinType>(): u64`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-fractional_part)
+* [`Libra::currency_code<CoinType>(): vector<u8>`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-currency_code)
+* [`Libra::lbr_exchange_rate<CoinType>(): FixedPoint32`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-lbr_exchange_rate)
+
+### Mutative ("setters")
+* [`Libra::update_lbr_exchange_rate<FromCoinType>`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-update_lbr_exchange_rate)
+* [`Libra::update_minting_ability<CoinType>`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-update_minting_ability)
+
+### Predicates
+* [`Libra::is_currency<CoinType>(): bool`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-is_currency)
+    - Related assertion: [`Libra::assert_is_currency<CoinType>()`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-assert_is_currency)
+* [`Libra::is_SCS_currency<CoinType>(): bool`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-is_SCS_currency)
+    - Related assertion: [`Libra::assert_is_SCS_currency<CoinType>()`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-assert_is_SCS_currency)
+* [`Libra::is_synthetic_currency<CoinType>(): bool`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-is_synthetic_currency)
+
+---
+# <a name="off_chain_use">Off-Chain Interactions</a>
```

Comment:
Why is this titled Off-Chain?

Reply:
This section is really talking about the events emitted for currencies and when they're emitted, so I think this isn't a good title for this section. I'll retitle the section to clarify this.